homeassistant/components/risco/sensor.py

```diff
@@ -88,7 +94,14 @@ def device_state_attributes(self):
         if self._event is None:
             return None
 
-        return {atr: getattr(self._event, atr, None) for atr in EVENT_ATTRIBUTES}
+        attrs = {atr: getattr(self._event, atr, None) for atr in EVENT_ATTRIBUTES}
+        if self._event.zone_id is not None:
+            zones = self.hass.data[DOMAIN][self._entry_id][DATA_ZONES]
+            zone = zones.get(self._event.zone_id)
+            if zone is not None:
+                attrs["zone_entity_id"] = zone.entity_id
```

Comment:
The entity is a platform concern and is not allowed to be accessed from outside its platform. Please refactor this.

One way is to use the dispatch helper to notify interested parties of changes to zone ids.